### DISCUSSION TITLE
PHP "missing" after PHP upgrade and downgrade

### DISCUSSION THREAD

**Discussion by bellwood** [Homebrew/discussions#2650] [Everyday usage] [answered]
I've been running php via `brew install php` for quite sometime now.

A recent upgrade brought me from PHP 8.0.x to 8.1.x which caused some issues requiring a downgrade.

I've successfully downgraded back, having to use PHP@8.0, however now I have packages saying that `php` is missing. 

Following most advice, I've already unlinked and relinked:

```
brew link --force --overwrite php@8.0
Warning: Already linked: /opt/homebrew/Cellar/php@8.0/8.0.13
To relink, run:
  brew unlink php@8.0 && brew link --force php@8.0
```

Relinking again doesn't resolve the issue `brew missing` still shows dependancies missing `PHP`.

I do have the PATHS set in my .zshrc

How do I go about resolving this and as an aside, is there a way to "lock" a package so that it will only upgrade for minors and not majors?

Thank you.

**Comment by gromgit** [comment, marked as answer]
If it's `brew missing` that's doing the complaining, that means you're missing the `php` _formula_, not the binary.
```
brew unlink php@8.0
brew install php
```
should fix that, but relinking `php@8.0` over `php` may break other PHP-based formulae that you've already installed.

If you only have one project that absolutely need PHP 8.0, simply reconfigure it to run `/opt/homebrew/opt/php@8.0/bin/php` instead of the unqualified `php`.

> is there a way to "lock" a package so that it will only upgrade for minors and not majors?

Not that I know of. `brew pin` will always pin the currently-installed version of a formula, so even minor upgrades will be ignored.

**Comment by bellwood** [comment]
> If it's brew missing that's doing the complaining, that means you're missing the php formula, not the binary.

Correct:
```[~] brew missing
composer: php
php-code-sniffer: php
```

...and they both seemingly work just fine.

I suppose the prudent move would be to get the project in question, Laravel, upgraded so that it supports PHP 8.1.

Thank you for your time and input.
